components/src/preact/MutationAnnotationsContext.spec.tsx

```diff
@@ -3,56 +3,114 @@ import { type FunctionalComponent } from 'preact';
+        );
+
         const { result } = renderHook(() => useMutationAnnotationsProvider(), { wrapper });
         return result.current;
     }
 
-    it('should return the correct annotation for a given nucleotide mutation', () => {
-        const result = renderAnnotationsHook()('A123', 'nucleotide');
+    describe('annotations for nucleotide mutations', () => {
+        const mockAnnotations: MutationAnnotations = [
+            {
+                name: 'Annotation 1',
+                description: 'Description 1',
+                symbol: 'A1',
+                nucleotideMutations: ['A123', 'A456'],
+            },
+            {
+                name: 'Annotation 2',
+                description: 'Description 2',
+                symbol: 'A2',
+                nucleotideMutations: ['A456', 'A789'],
+            },
+        ];
+
+        it('should return the correct annotation for a given mutation', () => {
+            const result = renderAnnotationsHook(mockAnnotations)(SubstitutionClass.parse('A123')!, 'nucleotide');
+
+            expect(result).toEqual([mockAnnotations[0]]);
+        });
+
+        it('should return the correct annotations if multiple contain a mutation', () => {
+            const result = renderAnnotationsHook(mockAnnotations)(SubstitutionClass.parse('A456')!, 'nucleotide');
 
-        expect(result).toEqual([mockAnnotations[0]]);
+            expect(result).toEqual([mockAnnotations[0], mockAnnotations[1]]);
+        });
     });
 
-    it('should return the correct annotations if multiple contain a mutation', () => {
-        const result = renderAnnotationsHook()('A456', 'nucleotide');
+    describe('annotations for amino acid mutations', () => {
+        const mockAnnotations: MutationAnnotations = [
+            {
+                name: 'Annotation 1',
+                description: 'Description 1',
+                symbol: 'A1',
+                aminoAcidMutations: ['B456', 'B789'],
+            },
+        ];
 
-        expect(result).toEqual([mockAnnotations[0], mockAnnotations[1]]);
+        it('should return the correct mutation annotation for a given mutations', () => {
+            const result = renderAnnotationsHook(mockAnnotations)(SubstitutionClass.parse('B456')!, 'amino acid');
+
+            expect(result).toEqual([mockAnnotations[0]]);
+        });
     });
 
-    it('should return undefined for a non-existent mutation code', () => {
-        const result = renderAnnotationsHook()('NON_EXISTENT', 'nucleotide');
+    describe('annotations for nucleotide positions', () => {
+        const mockAnnotations: MutationAnnotations = [
+            {
+                name: 'Annotation 1',
+                description: 'Description 1',
+                symbol: 'A1',
+                nucleotideMutations: ['A321T', 'A432T'],
+                nucleotidePositions: ['321', '543'],
+            },
+        ];
```

Comment:
I think this is missing a test:

```suggestion    describe('annotations for nucleotide positions', () => {
        const mockAnnotations: MutationAnnotations = [
            {
                name: 'Annotation 1',
                description: 'Description 1',
                symbol: 'A1',
                nucleotideMutations: ['A321T', 'A432T'],
                nucleotidePositions: ['321', '543'],
            },
            {
                name: 'Annotation 2',
                description: 'Description 2',
                symbol: 'A2',
                nucleotidePositions: ['432'],
            },
        ];

        it('should return both annotations if one matches the mutations and the other the position', () => {
            const result = renderAnnotationsHook(mockAnnotations)(SubstitutionClass.parse('A432T')!, 'nucleotide');
            expect(result).toEqual([mockAnnotations[0], mockAnnotations[1]]);
        });
```

Reply:
Good catch. Through this I also found a bug.